src/ripple/app/consensus/RCLConsensus.h

```diff
@@ -0,0 +1,377 @@
+        @param priorLedger Prior ledger used in consensus
+        @param believedCorrect Whether consensus believes currentLedger is LCL
+
+        @return The hash of the last closed network
+     */
+    uint256
+    getLCL (
+        uint256 const& currentLedger,
+        uint256 const& priorLedger,
+        bool believedCorrect);
+
+
+    /** Notification that the ledger has closed.
+
+       @param ledger the ledger we are changing to
+       @param haveCorrectLCL whether we believe this is the correct LCL
+    */
+    void
+    onClose(RCLCxLedger const & ledger, bool haveCorrectLCL);
+
+     /** Create our initial position of transactions to accept in this round
+         of consensus.
+
+          @param prevLedger The ledger the transactions apply to
+          @param isProposing Whether we are currently proposing
+          @param isCorrectLCL Whether we have the correct LCL
+          @param closeTime When we believe the ledger closed
+          @param now The current network adjusted time
+
+          @return Pair of (i)  transactions we believe are in the ledger
+                          (ii) the corresponding proposal of those transactions
+                               to send to peers
+     */
+    std::pair <RCLTxSet, typename RCLCxPeerPos::Proposal>
+    makeInitialPosition (
+        RCLCxLedger const & prevLedger,
+        bool isProposing,
+        bool isCorrectLCL,
+        NetClock::time_point closeTime,
+        NetClock::time_point now);
+
+
+    /** Dispatch a call to Consensus::accept
+
+        Accepting a ledger may be expensive, so this function can dispatch
+        that call to another thread if desired and must call the accept
+        method of the generic consensus algorithm.
+
+        @param txSet The transactions to accept.
+    */
+    void
+    dispatchAccept(RCLTxSet const & txSet);
+
+
+    /** Accept a new ledger based on the given transactions.
+
+        TODO: Too many arguments, need to group related types.
+
+        @param set The set of accepted transactions
+        @param consensusCloseTime Consensus agreed upon close time
+        @param proposing_ Whether we are proposing
+        @param validating_ Whether we are validating
+        @param haveCorrectLCL_ Whether we had the correct last closed ledger
+        @param consensusFail_ Whether consensus failed
+        @param prevLedgerHash_ The hash/id of the previous ledger
+        @param previousLedger_ The previous ledger
+        @param closeResolution_ The close time resolution used this round
+        @param now Current network adjsuted time
+        @param roundTime_ Duration of this consensus round
+        @param disputes_ Disputed trarnsactions from this round
+        @param closeTimes_ Histogram of peers close times
+        @param closeTime Our close time
+        @param json Json representation of consensus
+        @return Whether we should continue validating
+     */
+    bool
+    accept(
```

Comment:
For now, its "almost never" since the only caller is the generic consensus code itself and only in [one spot](https://github.com/bachase/rippled/blob/5aa6bda21d6446f2756b17142db9c29039cc4e7d/src/ripple/consensus/Consensus.h#L1727)